pkg/objstore/oci/oci.go

```diff
@@ -0,0 +1,331 @@
+	}
+	return response.Content, nil
+}
+
+// Upload the contents of the reader as an object into the bucket.
+// Upload should be idempotent.
+func (b *Bucket) Upload(ctx context.Context, name string, r io.Reader) error {
+	return uploadObject(ctx, *b, name, r)
+}
+
+// Exists checks if the given object exists in the bucket.
+func (b *Bucket) Exists(ctx context.Context, name string) (bool, error) {
+	_, err := getObject(ctx, *b, name, "")
+	if err != nil {
+		if b.IsObjNotFoundErr(err) {
+			return false, nil
+		}
+		return false, errors.Wrapf(err, "cannot get OCI object '%s'", name)
+	}
+	return true, nil
+}
+
+// Delete removes the object with the given name.
+// If object does not exists in the moment of deletion, Delete should throw error.
+func (b *Bucket) Delete(ctx context.Context, name string) error {
+	return deleteObject(ctx, *b, name)
+}
+
+// IsObjNotFoundErr returns true if error means that object is not found. Relevant to Get operations.
+func (b *Bucket) IsObjNotFoundErr(err error) bool {
+	failure, isServiceError := common.IsServiceError(err)
+	if isServiceError {
+		k := failure.GetHTTPStatusCode()
+		match := k == http.StatusNotFound
+		level.Debug(b.logger).Log("msg", match)
+		return failure.GetHTTPStatusCode() == http.StatusNotFound
+	}
+	return false
+}
+
+// ObjectSize returns the size of the specified object.
+func (b *Bucket) ObjectSize(ctx context.Context, name string) (uint64, error) {
+	response, err := getObject(ctx, *b, name, "")
+	if err != nil {
+		return 0, err
+	}
+	return uint64(*response.ContentLength), nil
+}
+
+// Close closes bucket.
+func (b *Bucket) Close() error {
+	return nil
+}
+
+// Attributes returns information about the specified object.
+func (b *Bucket) Attributes(ctx context.Context, name string) (objstore.ObjectAttributes, error) {
+	response, err := getObject(ctx, *b, name, "")
+	if err != nil {
+		return objstore.ObjectAttributes{}, err
+	}
+	return objstore.ObjectAttributes{
+		Size:         *response.ContentLength,
+		LastModified: response.LastModified.Time,
+	}, nil
+}
+
+// createBucket creates bucket.
+func (b *Bucket) createBucket(ctx context.Context, compartmentId string) (err error) {
+	return createBucket(ctx, b, compartmentId)
+}
+
+// deleteBucket deletes bucket.
+func (b *Bucket) deleteBucket(ctx context.Context) (err error) {
+	return deleteBucket(ctx, b)
+}
+
+// NewBucket returns a new Bucket using the provided oci config values.
+func NewBucket(logger log.Logger, ociConfig []byte) (*Bucket, error) {
+	level.Debug(logger).Log("msg", "creating new oci bucket connection")
+	var config = DefaultConfig
+	var configurationProvider common.ConfigurationProvider
+	var err error
+
+	if err := yaml.Unmarshal(ociConfig, &config); err != nil {
+		return nil, errors.Wrapf(err, "unable to unmarshal the given oci configurations")
+	}
+
+	provider := Provider(strings.ToLower(config.Provider))
+	level.Info(logger).Log("msg", fmt.Sprintf("Creating OCI default '%s' provider", provider))
```

Comment:
Nit: the pattern is to specify separate fields as separate arguments so it should be:
```suggestion
	level.Info(logger).Log("msg", "creating OCI client", "provider", provider)
```

Reply:
Changed.